src/object/srv_obj.c

```diff
@@ -648,7 +959,7 @@ ds_obj_rw_local_hdlr(crt_rpc_t *rpc, uint32_t tag, struct ds_cont_hdl *cont_hdl,
 		rc = vos_update_begin(cont->sc_hdl, orw->orw_oid,
 				      orw->orw_epoch, &orw->orw_dkey,
 				      orw->orw_nr, orw->orw_iods.ca_arrays,
-				      ioh);
+				      ioh, dth, &dti_cos_count, dti_cos);
```

Comment:
That will cause additional PMDK transaction.

Reply:
> If this dkey hash would be only needed for this resend case, since this happens rarely, so probably do not add burden to RPC in this case?

The dkey_hash also will be used by DTX for CoS cache related logic.

Reply:
That will cause additional PMDK transaction.

Hmm, even inside vos_update_begin(), vos_dtx_commit happens before PMDK transaction start?

Reply:
Unless the operation really hit conflict DTX (for vos_obj_hold) that will cause the CoS DTXs committed when vos_update_begin(), otherwise, the CoS DTXs will be committed during the vos_update_end() that will share the PMDK transaction with the modification.

Reply:
ah, this is because of vos_obj_hold() inside vos_update_begin() ?  hmm, we probably need discuss VOS layer in beijing meeting. The interface is not very clear to me.

Reply:
yes